openid4vc-high-assurance-interoperability-profile-1_0.md

```diff
@@ -343,6 +343,78 @@ Note: The issuer MAY decide to support both options. In which case, it is at the
 
 A Credential Format Profile for Credentials complying with IETF SD-JWT VCs [@!I-D.ietf-oauth-sd-jwt-vc] is defined in Annex A.3 of [@!OIDF.OID4VCI] and Annex A.4 of [@!OIDF.OID4VP].
 
+## SD-JWT VC Data Model (SD-JWT VCDM)
+
+SD-JWT VCDM is a credential format that is compliant to IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] that allows the usage of existing data structures represented using [@!W3C.VCDM1.1] or [@!W3C.VCDM2.0], while enabling selective disclosure.
+
+SD-JWT VCDM credentials contain a data structure that can be processed using a JSON-LD processor after applying SD-JWT processing.
+
+When IETF SD-JWT VC is mentioned in this specification, SD-JWT VCDM defined in this section MAY be used.
+
+For backward compatibility with JWT processors, the following registered JWT claims MUST be used, instead of their respective counterpart properties in [@!W3C.VCDM1.1] or [@!W3C.VCDM2.0]:
+
+* `exp` Claim MUST represent the `expirationDate` property, encoded as a UNIX timestamp (NumericDate).
+* `iss` Claim MUST represent the `issuer` property.
+* `iat` Claim MUST represent `issuanceDate` property, encoded as a UNIX timestamp (NumericDate).
+* `status` Claim MUST represent `credentialStatus` property.
+* `sub` Claim MUST represent the `id` property of `credentialSubject` property.
+
+IETF SD-JWT VC is extended with the following claims:
+
+* `vcdm`: OPTIONAL. Contains properties defined in [@!W3C.VCDM1.1] or [@!W3C.VCDM2.0] that are not represented by their counterpart JWT Claims as defined above.
```

Comment:
How is `"vcdm"` different from the good old [`"vc"` claim](https://www.w3.org/TR/vc-data-model/#iana-considerations)?

Reply:
I recommend removing the "vcdm" if the governance of the data model is well defined. (see my question on how to express claims from different vocabularies)

Reply:
The `vcdm` claim is there to allow JSON-LD processors to process the `vcdm` object which has to be a valid JSON-LD object in compact form. Some JSON-LD processors would have issues with terms that are not included in a context definition. There might be some SD-JWT and SD-JWT VC claims today or in the future that are not included and therefore cause issues. By encapsulating the JSON-LD stuff within `vcdm`, we guarantee this won't cause any issues.

Reply:
> By encapsulating the JSON-LD stuff within `vcdm`

Isn't that exactly the same reason why you introduced `vc` a few years ago? Why not re-use that?

Reply:
I guess `vc` would be fine, but it is an IANA-registered claim that's defined for use in a different context. Might be okay to re-use though. `vcdm` would make it clear that this claim is to be used with the processing model in mind that @awoie describes in one of the other suggested changes.

Reply:
- it does not repeat the errors from the past because vcdm is much better defined in the context of sd-jwt vcdm. 
- it could be `vc` claim, personally think `vcdm` is cleaner because IANA reserved `vc` for w3c vcdm 1.1. 
- we could probably do without `vcdm` at all, but processing using different processors is slightly easier when `vcdm` claim is present, so would prefer to keep it. 

please re-read the above paragraphs on the purpose of sd-jwt vcdm - what you propose in your third bullet point there is not what sd-jwt vcdm is for; for that, please just use w3c vcdm.

Reply:
VCDM should not be there as there's no need for it if we follow the processing logic above (which worked well with the existing use cases, creating a clear boundary and positioning the JSON-LD so that it can be used whenever needed)

Reply:
> * IANA reserved `vc` for w3c vcdm 1.1.

This raises the question whether `vcdm` is  only for VCDM 1.1 or for both VCDM 1.1 and VCDM 2.0 ?

Reply:
vcdm is intended for both VCDM 1.1 and VCDM 2.0 - I thought text made it pretty clear?

Reply:
'vcdm' should not be there as there's no need for it if we follow the processing logic above (which worked well with the existing use cases, creating a clear boundary and positioning the JSON-LD so that it can be used whenever needed)

Reply:
I think `vc` won't be used. It would be `"vcdm": { VCDM 1.1 or VCDM 2.0 stuff }`.

Reply:
Why you need that?

See:
- base data format is JSON
- if use cases want to use JSON-LD, JSON-LD should cover all the definitions needed to successfully process the data model (again, depending on for what purpose JSON-LD is used)
- JSON-LD context MUST be treated as any referenced content in the data model (integrity protection, ability to share the data as unsigned payload);

Why this cannot work?

Reply:
Perhaps this is my lack of real world JSON-LD processors but wouldn't this require that all things that are not part of the JSON-LD VCDM context have to be defined in proper JSON-LD context to not cause issues with these JSON-LD processors, e.g., `status`, `vct`?

Reply:
Yes, and the proposal is that if JSON-LD is used, the JSON-LD context must cover all the claims. It leads to a very clean design without exception handling.

Reply:
If implementations had issues
a) what are the issues?
b) can we contact them and learn more about the issues?

Reply:
> I think `vc` won't be used. It would be `"vcdm": { VCDM 1.1 or VCDM 2.0 stuff }`.

But @Sakurann said:

> IANA reserved `vc` for w3c vcdm 1.1

Wouldn't there be a lot of ambiguity and interop problems if you define two different JWT claims that can be used for the same purpose (VCDM 1.1)?

Reply:
I don't see how that would be a problem. The SD-JWT VC DM wrapper is new and has to be implemented anyway; any claim name would work. What is important is what is in the `vcdm` claim.

Reply:
> any claim name would work

So are you saying if an Issuer wants to use VCDM 1.1 wrapped inside a JWT, they could use either the `vc` or `vcdm` claim, since both serve the same purpose? And would Verifiers also be expected to support both? Or pick one of the two?

Reply:
I meant to say that we can define any claim name here, without much impact on implementations. Whether they process `vcdm`, `vc` or `foo` doesn't matter. There may be a preference of one over the other for historic reasons, or because one has an existing IANA definition, or because one has clearer semantics, but other than that, it doesn't matter.